### PR TITLE
GH-15115: [R][CI] pyarrow tests fail on macos 10.13 due to missing pyarrow wheel

### LINKED ISSUE #15115: [R][CI] pyarrow tests fail on macos 10.13 due to missing pyarrow wheel
### Describe the bug, including details regarding any error messages, version, and platform.

see https://github.com/apache/arrow/issues/14829#issuecomment-1367506270

### Component(s)

Continuous Integration, R

### PR DESCRIPTION
**Pull request by assignUser**
* Closes: #15115

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit homebrew-r-autobrew

File: r/tests/testthat/test-python.R [hunk 1]
```diff
@@ -24,6 +24,8 @@ test_that("install_pyarrow", {
   skip_if_not_installed("reticulate")
   # PyArrow doesn't support Python 3.6 or earlier
   skip_on_python_older_than("3.7")
+  # no pyarrow wheels for macos 10.13
+  skip_if(Sys.info()["sysname"] == "Darwin" && Sys.info()["release"] < 14)
```
Comment (jonkeane) [inline r/tests/testthat/test-python.R:28]: IIUC the release here should actually be 18 (cf https://en.wikipedia.org/wiki/Darwin_(operating_system))

**Review by jonkeane — CHANGES_REQUESTED** [review]
I suspect this will still fail, but that might be the head issue we were seeing in the other PR.

File: r/tests/testthat/test-python.R (hunk 1 above)
Comment (assignUser) [reply r/tests/testthat/test-python.R:28]: oh wow that versioning scheme is confusing :D

**Comment by assignUser** [comment]
@github-actions crossbow submit homebrew-r-autobrew

and yes the head() issue will probably still pop up but the pyarrow tests should all be skipped like this (the first explicitly and the others due to missing pyarrow)

**Review by jonkeane — APPROVED** [review]

**Pull request merged**